qt/widgets/common/src/WorkspaceSelector.cpp

```diff
@@ -277,6 +287,24 @@ bool WorkspaceSelector::hasValidSuffix(const QString &name) const {
   return false;
 }
 
+bool WorkspaceSelector::hasValidNumberOfBins(
+    Mantid::API::Workspace_sptr object) const {
+  if (m_binLimits.first != 0 || m_binLimits.second != -1) {
+    auto const group =
+        boost::dynamic_pointer_cast<Mantid::API::WorkspaceGroup>(object);
+    if (group == nullptr) {
```

Comment:
It's possible that the workspace is not a `MatrixWorkspace` either. I think this would be safer if it just did
```c++
if(auto matrixWS = boost::dynamic_pointer_cast<Mantid::API::MatrixWorkspace>(object)) {
... bin number checking code
}